### ISSUE TITLE
Add annotations to manifest from within app

### ISSUE THREAD

**Issue by karawoo** [Sage-Bionetworks/dccvalidator#61]
One thing we've gone back and forth on in AMP-AD is whether to make people redundantly add the same metadata in both the metadata files and manifest. One possibility is that the app could accept the metadata files and manifest with just path, parent, and specimen ID, then the app can merge the manifest with the annotations from the metadata files and provide the result as a file for download. The user would then use this new manifest to upload data (without having to fill out all the annotations themselves twice).

**Comment by kelshmo** [comment]
I would help implement this!

**Comment by karawoo** [comment]
That would be awesome @kelshmo!

**Comment by kelshmo** [comment]
Note: multiple annotations are required for this use case to be fully implemented.

**Comment by karawoo** [comment]
I could also see this as a feature of the dccmonitor tool -- it could create a manifest that curators could use to update annotations once files are uploaded by the contributor.

**Comment by kelshmo** [comment]
Can we have a design meeting to talk through different approaches?

**Comment by karawoo** [comment]
Sorry I missed this comment -- sure, send @Aryllen and me a calendar invite if you want to discuss!

**Comment by kelshmo** [comment]
@karawoo and I discussed the utility of a feature in the `dccmonitor` that would programmatically join a manifest and assay metadata file by `biospecimen` and `assay`, then the biospecimen file by `specimenID` and finally the individual metadata file by `individualID` to update annotations on a recently uploaded file. As the metadata files get annotated by `metadataType` when they are uploaded to the validator, a function could crawl the file hierarchy and order these files appropriately to enable a join.

We discussed this join as taking place on the back-end for the curators in place of adding a step for the users to download a joined manifest.

We also discussed adding a step that flags these metadata files as complete or validated to further automate this step.
